apiserver/pkg/client/kubernetes.go

```diff
@@ -35,66 +31,16 @@ func (c *KubernetesClient) ConfigMapClient(namespace string) v1.ConfigMapInterfa
 
 // CreateKubernetesCoreOrFatal creates a new client for the Kubernetes pod.
 func CreateKubernetesCoreOrFatal(initConnectionTimeout time.Duration, options util.ClientOptions) KubernetesClientInterface {
-	var client KubernetesClientInterface
-	var err error
-	var operation = func() error {
-		// In cluster
-		kubernetesClient, err := newInClusterKubernetesClient(options)
-		if err != nil {
-			return err
-		}
-		client = kubernetesClient
-		return nil
-	}
-
-	// Out of cluster
-	kubernetesClient, err := newOutOfClusterKubernetesClient()
-	if err == nil {
-		return kubernetesClient
-	}
-	klog.Infof("(Expected when in cluster) Failed to create Kubernetes client by out of cluster kubeconfig. Error: %v", err)
-
-	klog.Infof("Starting to create Kubernetes client by in cluster config.")
-	b := backoff.NewExponentialBackOff()
-	b.MaxElapsedTime = initConnectionTimeout
-	if err := backoff.Retry(operation, b); err != nil {
-		klog.Fatalf("Failed to create pod client. Error: %v", err)
-	}
-
-	return client
-}
-
-func newInClusterKubernetesClient(options util.ClientOptions) (*KubernetesClient, error) {
-	restConfig, err := rest.InClusterConfig()
+	cfg, err := config.GetConfig()
 	if err != nil {
-		return nil, errors.Wrap(err, "Failed to initialize kubernetes client.")
+		glog.Fatalf("Failed to create TokenReview client. Error: %v", err)
```

Comment:
The precedence changes a little bit.  `config.GetConfig()`  checks `--kubeconfig` first and then `in-cluster` config. I can not remember why I choose to check `in-cluster` first and then external. The change should work for us. 

Config precedence after the change. 
```
* --kubeconfig flag pointing at a file
* KUBECONFIG environment variable pointing at a file
* In-cluster config if running in cluster
* $HOME/.kube/config if exists.
```